modules/snooker/src/main/ts/ephox/snooker/operate/Redraw.ts

```diff
@@ -15,31 +15,43 @@ interface NewRowsAndCells {
   readonly newCells: SugarElement[];
 }
 
-const render = function <T extends DetailNew> (table: SugarElement, grid: RowDataNew<T>[]): NewRowsAndCells {
+const render = <T extends DetailNew> (table: SugarElement, grid: RowDataNew<T>[]): NewRowsAndCells => {
   const newRows: SugarElement[] = [];
   const newCells: SugarElement[] = [];
 
-  const insertThead = Arr.last(SelectorFilter.children(table, 'caption,colgroup')).fold(
-    () => Fun.curry(Insert.prepend, table),
-    (c) => Fun.curry(Insert.after, c)
-  );
+  const insert = (selector: string, element: SugarElement<HTMLTableSectionElement | HTMLTableColElement>) => {
+    const lastChild = Arr.last(SelectorFilter.children(table, selector));
 
-  const renderSection = function (gridSection: RowDataNew<T>[], sectionName: 'thead' | 'tbody' | 'tfoot') {
-    const section = SelectorFind.child(table, sectionName).getOrThunk(function () {
+    const foldfunction = lastChild.fold(
+      () => Fun.curry(Insert.prepend, table),
```

Comment:
The `Fun.curry` is really no longer needed anymore. That was only done because it was a static function and we didn't have access to element. As such this could be simplified now by just calling the `prepend` or `after` functions directly.

Reply:
The implementation you changed this to is incorrect/not the same as the current functionality. Please see the diff I sent you earlier today, as your current version will place the colgroup/caption outside the table and never insert the thread.